### PR TITLE
Correct implementation of _MCRem

### PR DESCRIPTION
**Pull request by AtariDreams**
Removed useless operations

Rewrote parts of the functions to better align with the algorithms they are based on

Better codegen for places were we do modulo

### CODE REVIEW

**Comment by AtariDreams** [comment]
This doesn't work because qh is reassigned anyway. So the assignment does nothing.

**Comment by fsb4000** [comment]
Yes, I agree. 
godbolt.org generates the same assembly with `--qh;` and without `--qh;`.

But it generates 8 extra instructions with `unsigned int`: https://godbolt.org/z/cj56a4h6s

File: stl/src/multprec.cpp [hunk 1]
```diff
@@ -63,16 +63,16 @@ void __CLRCALL_PURE_OR_CDECL _MP_Mul(
 
     // Knuth, vol. 2, p. 268, Algorithm M
     // M1: [Initialize.]
-    for (int i = 0; i < m + n + 1; ++i) {
+    for (unsigned int i = 0; i < m + n + 1; ++i) {
```
Comment (StephanTLavavej) [inline stl/src/multprec.cpp:66]: The correctness of the `int` => `unsigned int` change seems reasonably easy to verify, but I don't believe that this is a clear improvement, and @fsb4000 has found potential negative codegen impact. Although I have a slight preference for unsigned types in general, I don't see sufficient value in changing this existing code, so I believe these changes should be dropped.

File: stl/src/multprec.cpp [hunk 2]
```diff
@@ -170,7 +170,6 @@ void __CLRCALL_PURE_OR_CDECL _MP_Rem(
         }
         // D5: [Test remainder.]
         if (k != 0) { // D6: [Add back.]
-            --qh;
```
Comment (StephanTLavavej) [inline stl/src/multprec.cpp:173]: No change requested, noting for other reviewers: This is safe because `qh` is a loop-local variable on line 131.

**Review by StephanTLavavej — CHANGES_REQUESTED** [review]

**Comment by StephanTLavavej** [comment]
@AtariDreams We've asked you a few times to please avoid force-pushing commits after review has started, as this makes it harder to see what's changed. Is there some tool or part of your workflow that's doing this?

**Comment by AtariDreams** [comment]
No, I apologize. I will refrain from doing this.

**Comment by AtariDreams** [comment]
Okay so I was able to reduce the number of instructions even further from the original

**Comment by AtariDreams** [comment]
@StephanTLavavej Ready!

**Comment by AtariDreams** [comment]
Since the focus of this PR has changed. I will close it and reopen it as a new one.

**Pull request closed without merging**